src/features/island/buildings/components/building/petHouse/FeedPets.tsx

```diff
@@ -0,0 +1,240 @@
+    if (!isBulkFeed) {
+      setIsBulkFeed(true);
+      const newSelectedFeed: {
+        pet: PetName | number;
+        food: CookableName;
+      }[] = [];
+
+      // Create a map to track how much of each food we've allocated
+      const foodAllocation: Partial<Record<CookableName, number>> = {};
+
+      // First pass: collect all food requests and count available inventory
+      const foodRequests: Array<{
+        petName: PetName | number;
+        food: CookableName;
+      }> = [];
+      activePets.forEach(([petName, pet]) => {
+        if (pet) {
+          pet.requests.food.forEach((food) => {
+            const isAlreadyFed = isFoodAlreadyFed(pet, food);
+            if (!isAlreadyFed) {
+              foodRequests.push({ petName, food });
+              if (!foodAllocation[food]) {
+                foodAllocation[food] = 0;
+              }
+            }
+          });
+        }
+      });
+
+      // Second pass: select food items based on available inventory
+      foodRequests.forEach(({ petName, food }) => {
+        const availableFood = inventory[food] ?? new Decimal(0);
+        const currentAllocation = foodAllocation[food] || 0;
+
+        // Only select if we have enough food available
+        if (availableFood.greaterThan(currentAllocation)) {
+          newSelectedFeed.push({ pet: petName, food });
+          foodAllocation[food] = currentAllocation + 1;
+        }
+      });
+
+      setSelectedFeed(newSelectedFeed);
+    } else {
+      handleConfirmFeed();
+    }
+  }, [isBulkFeed, activePets, inventory, handleConfirmFeed]);
+
+  // Memoize the mapped pets calculation
+  const mappedPets = useMemo(() => {
+    return selectedFeed.reduce<
+      {
+        pet: PetName | number;
+        food: CookableName[];
+      }[]
+    >((acc, { pet, food }) => {
+      const existingPet = acc.find((p) => p.pet === pet);
+      if (existingPet) {
+        existingPet.food.push(food);
+      } else {
+        acc.push({ pet, food: [food] });
+      }
+      return acc;
+    }, []);
+  }, [selectedFeed]);
+
+  // Memoize the cancel handler
+  const handleCancel = useCallback(() => {
+    setSelectedFeed([]);
+    setIsBulkFeed(false);
+  }, []);
+
+  return (
+    <>
+      <InnerPanel className="flex flex-row justify-between mb-1 p-1 gap-1">
+        <div className="flex flex-col sm:flex-row items-center gap-1">
+          <Label type={isBulkFeed ? "vibrant" : "formula"}>
+            {isBulkFeed ? `Bulk Feed Mode` : `Your Pets (${activePets.length})`}
+          </Label>
+          {isBulkFeed && (
+            <Label type={"warning"}>
+              {`Feed Selected: ${selectedFeed.length}`}
+            </Label>
+          )}
+        </div>
+        <div className="flex flex-row gap-2">
+          <Button
+            className="w-40"
+            disabled={isBulkFeed && selectedFeed.length === 0}
+            onClick={handleBulkFeed}
+          >
+            {isBulkFeed ? `Confirm Feed` : `Bulk Feed`}
```

Comment:
Another hard coded string - there are a few in this file that will need to use app translation to ensure it works in other languages 👍